### PR TITLE
Adding tests for user registration

### LINKED ISSUE #8: Add Testcases for Registration
**Is your feature request related to a problem? Please describe.**

Taking [Teesco Backend](https://github.com/ECellNitrr/teesco-backend/) as a reference, test the User Registration API.

**Describe the solution you'd like**

Create a file in `/tests/users` named `tests_register_api.py` and create testcases for the API.

**Additional context**

https://github.com/ECellNitrr/teesco-backend/blob/dev/tests/users/tests_register_api.py

### PR DESCRIPTION
**Pull request by akhil-maker**
### Description
Test cases are added for user registration using various cases:
                                              - test_with_correct_credentials
                                              - invalid details
                                              - blank details
                                              - used email(registered)
                                              - small, invalid and blank password
                                              - without required fields

**Type of Change:** (Delete irrelevant options)
- Testing

**Code/Quality Assurance Only** (Delete irrelevant options)

- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
Tested locally and checked that all test cases are successful.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by akhil-maker** [comment]
@anshsrtv and @priyanshisharma  
please give your reviews!!

**Review by priyanshisharma — APPROVED** [review]
Looks good to me 🚀 
Try making a new branch (Pull from `upstream master` in your fork's `master` and then checkout to a new branch and work there) for each PR instead of making PRs from master branch, this shall reduce the number of commits that aren't relevant to the PR.

File: tests/users/test_user_register.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        data = {
+            "first_name": "string1",
+            "last_name": "string2",
+            "email": "user234@example.com",
+            "contact": "string9000",
+            "password": "string989"
+        }
+
+        response = self.client.post("/users/register/", data)
+        response = self.client.post("/users/register/", data)
```
Comment (anshsrtv) [inline tests/users/test_user_register.py:27]:
```suggestion
```

File: tests/users/test_user_register.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_fail_without_fields(self):
```
Comment (anshsrtv) [inline tests/users/test_user_register.py:30]:
```suggestion
    def test_fail_without_lastname(self):

```

**Review by anshsrtv — CHANGES_REQUESTED** [review]
Nicely done 💯 Just make the suggested changes and delete `users/tasks.py` and `users/forms.py` if they aren't being used anywhere.

File: tests/users/test_user_register.py (hunk 1 above)
Comment (akhil-maker) [reply tests/users/test_user_register.py:27]: @anshsrtv sir this is used first time so as to register it as a new user and next time it is used so as to generate 400 bad request error for already registered or used email.

File: tests/users/test_user_register.py (hunk 2 above)
Comment (akhil-maker) [reply tests/users/test_user_register.py:30]: Ok! I will change it.

File: tests/users/test_user_register.py (hunk 1 above)
Comment (anshsrtv) [reply tests/users/test_user_register.py:27]: Okay cool 😆

Comment (akhil-maker) [reply tests/users/test_user_register.py:27]: > Nicely done 💯 Just make the suggested changes and delete `users/tasks.py` and `users/forms.py` if they aren't being used anywhere.

@anshsrtv sir, 
I don't find the usage of tasks.py anywhere so i will delete it but forms.py is used in admin.py so please can you tell me whether to add or delete.

**Comment by akhil-maker** [comment]
> Nicely done 💯 Just make the suggested changes and delete `users/tasks.py` and `users/forms.py` if they aren't being used anywhere.

@anshsrtv sir,
I don't find the usage of tasks.py anywhere so i will delete it but forms.py is used in admin.py so please can you tell me whether to add or delete.

**Comment by anshsrtv** [comment]
Don't delete it then.

**Review by anshsrtv — APPROVED** [review]

**Pull request merged**